flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -181,43 +175,26 @@
 	 */
 	@Test
 	public void testEarlyCanceling() throws Exception {
-		Deadline deadline = Deadline.fromNow(Duration.ofMinutes(2));
-		StreamConfig cfg = new StreamConfig(new Configuration());
+		final StreamConfig cfg = new StreamConfig(new Configuration());
 		cfg.setOperatorID(new OperatorID(4711L, 42L));
 		cfg.setStreamOperator(new SlowlyDeserializingOperator());
 		cfg.setTimeCharacteristic(TimeCharacteristic.ProcessingTime);
 
-		Task task = createTask(SourceStreamTask.class, cfg, new Configuration());
+		final TaskManagerActions taskManagerActions = mock(TaskManagerActions.class);
```

Comment:
Agree.

Reply:
I introduce a `NoOpTaskManagerActions` (in fact move an existing inner class to public) and `spy` on it. Because here we are mainly concerned about a specific function call happened, `verify` is the natural way IMO. Otherwise we might use a blocking queue and verify the message queue, more complex without additional benefit.